superset-frontend/packages/superset-ui-core/src/connection/callApi/parseResponse.ts

```diff
@@ -58,17 +58,17 @@ export default async function parseResponse<T extends ParseMethod = 'json'>(
     const result: JsonResponse = {
       response,
       json: cloneDeepWith(json, (value: any) => {
-        // `json-bigint` could not handle floats well, see sidorares/json-bigint#62
-        // TODO: clean up after json-bigint>1.0.1 is released
-        if (value?.isInteger?.() === false) {
-          return Number(value);
-        }
         if (
           value?.isGreaterThan?.(Number.MAX_SAFE_INTEGER) ||
           value?.isLessThan?.(Number.MIN_SAFE_INTEGER)
         ) {
           return BigInt(value);
         }
+        // // `json-bigint` could not handle floats well, see sidorares/json-bigint#62
+        // // TODO: clean up after json-bigint>1.0.1 is released
+        if (value?.isNaN?.() === false) {
```

Comment:
@justinpark Won't decimal numbers be truncated in case they are passed to the native Javascript `BigInt` object? This can happen when `value?.isGreaterThan?.(Number.MAX_SAFE_INTEGER) || value?.isLessThan?.(Number.MIN_SAFE_INTEGER)`. In other words, do we need to include a check for type integer?
```
if (
   value?.isInteger?.()  <-- check for integer
   value?.isGreaterThan?.(Number.MAX_SAFE_INTEGER) ||
   value?.isLessThan?.(Number.MIN_SAFE_INTEGER)
) {
   return BigInt(value);
}
```

Also, if that's the case, don't we also need to handle large decimals given that `toNumber` will truncate?

Reply:
Adding an integer check to BigInt is a good idea. I have added that part.

I think the truncation of large decimals is a trade-off. For example, when a value used as bigint is truncated, it affects the accuracy of the information related to the ID, so it is appropriate to keep all decimals by bigint format. However, in the case of large decimal floating numbers, since the values are within a range that allows arithmetic operations, I believe it is a part that can be compromised to support operations in such cases, even if some decimal can be truncated.